pkg/cmd/util/clientcmd/gating.go

```diff
@@ -1,64 +1,49 @@
 package clientcmd
 
 import (
-	"encoding/json"
 	"fmt"
 
-	"github.com/blang/semver"
-
 	"github.com/openshift/origin/pkg/client"
-	"github.com/openshift/origin/pkg/version"
 )
 
-// Gate returns an error if the server is below minServerVersion or above/equal maxServerVersion.
-// To test only for min or only max version, set the other string to the empty value.
-func Gate(ocClient *client.Client, minServerVersion, maxServerVersion string) error {
-	if len(minServerVersion) == 0 && len(maxServerVersion) == 0 {
-		return fmt.Errorf("No version info passed to gate command")
+// Return err if the server does not support any of the legacy policy objects (< 3.7)
+func LegacyPolicyResourceGate(ocClient *client.Client) error {
+	return ResourceGate(ocClient, "authorization.openshift.io/v1", []string{
+		"clusterpolicies",
+		"clusterpolicybindings",
+		"policies",
+		"policybindings",
+	})
+}
+
+// Do a discovery for server resources of the provided groupVersion, returning nil if any one of resourceNames is supported
+// by the server.
```

Comment:
I would think that we should check if *all* of the resources are availble, not any, as in most case API are not actually interchangable. Otherwise we force callers to only pass one resource at a time and make multiple calls to make sure all the resources they need are available.

Reply:
Bonus points would be if we return the list of unavailable ones with the error code, when some are unavailable.

Reply:
I suppose it could do `[]GVR, err` where `err != nil` is always fatal, and then `LegacyPolicyResourceGate` could wrap that with and extra error on a non-empty `[]GVR`.